### ISSUE TITLE
Chained batch should be persistent

### ISSUE THREAD

**Issue by vweevers** [Level/leveldown#620]
A segfault can happen when a chained batch is GC-ed in between `batch.write(callback)` and `BatchWriteWorker->DoExecute()`.

**Comment by vweevers** [comment]
Wrote a test and ran it in v4 as well. Only v5 has the bug.

**Comment by peakji** [comment]
I've obtained a core dump file on segfault **11/SEGV** from a server running v5.0.0 (been waiting for a week!).

Back trace shows the segment fault happened on a uv worker pool thread, but not sure if it is running `BatchWriteWorker->DoExecute()` or not.

However, I found one unwritten batch which means the software crashed while writing batches. Since the software does not perform other operations during this phase, I think it is safe to say this PR fixed the segfault described in https://github.com/Level/leveldown/pull/601. 🎉

I'll deploy v5.0.3 today and see if it crashes.

----

Back trace generated by `lldb` & `llnode`:

<img width="731" alt="backtrace" src="https://user-images.githubusercontent.com/219051/56849362-fd8b9300-6925-11e9-8ea5-ce8f12894d3c.png">

Objects:

<img width="959" alt="objects" src="https://user-images.githubusercontent.com/219051/56849365-02504700-6926-11e9-94a1-44b4a7b655f8.png">

**Comment by vweevers** [comment]
@peakji that's good! Your analysis makes sense. I assume you're running parallel batches? The presence of that unwritten batch seems to indicate GC did not collect it yet, so the segfault must have been caused by another batch (perhaps those "Invalid values"?).

I also deployed 5.0.3 a few hours ago. Ended up using chained batch. Low volume atm, increasing on monday. In a week or so, our use cases combined should give us all the confidence we need 💯
